R/ResultFlagsDependent.R

```diff
@@ -548,6 +548,126 @@ TADA_FindQCActivities <- function(.data, clean = FALSE, flaggedonly = FALSE) {
 }
 
 
+#' Pair Replicates with Original Samples
+#' 
+#' This function looks for replicate samples and pairs them to their original or
+#' parent sample for further analysis. Replicate samples without an associated original
+#' sample are flagged as orphan samples.
+#' 
+#' @param .data TADA dataframe which must include the columns 'OrganizationIdentifier', 'ActivityTypeCode',
+#' 'ActivityStartDate', 'ActivityStartTime.Time', 'TADA.LatitudeMeasure', 'TADA.LongitudeMeasure',
+#' 'TADA.ResultMeasureValue', 'TADA.ComparableDataIdentifier', 'TADA.ActivityType.Flag', 
+#' and 'TADA.ActivityDepthHeightMeasure.MeasureValue'.
+#' @param type Character argument identifying which Activity Types to look for while pairing replicates
+#' to their parent samples. The default type is "QC_replicate", which includes Activity Type Codes: 
+#' "Quality Control Field Replicate Habitat Assessment",
+#' "Quality Control Field Replicate Msr/Obs",
+#' "Quality Control Field Replicate Portable Data Logger",
+#' "Quality Control Field Replicate Sample-Composite", and
+#' "Quality Control Sample-Field Replicate".
+#' @param time_window Numeric argument defining the time window in seconds within 
+#' which to search for parent samples. The default time window is 300 seconds or 5 minutes.
+#' 
+#' @return This function adds two columns to the original data frame: 'TADA.DateTime' and 
+#' 'TADA.ReplicateSampleID'. 'TADA.DateTime' combines the columns 'ActivityStartDate' and 
+#' 'ActivityStartTime.Time' into one date-time of class POSIXct to enable the time window 
+#' search for parent samples. 'TADA.ReplicateSampleID' contains the 'ResultIdentifier' value 
+#' from the replicate sample if a parent sample match is identified. Both the replicate sample
+#' and the parent sample will have the same 'ResultIdentifier' code in this column, marking them
+#' as a pair. If a sample was identified as a replicate sample in the 'TADA.ActivityType.Flag' 
+#' column but does not have an associated parent sample in the data frame, the 'TADA.ReplicateSampleID'
+#' column will contain the flag 'Orphan'. If more than one parent sample is identified in the
+#' data frame, the 'TADA.ReplicateSampleID' column will contain the flag 'Further Review', and 
+#' it may be necessary to rerun the analysis with a more narrow time window.
+#' 
+#' @export
+#' 
+#' @examples 
+#' # Load example dataset:
+#' data(Data_NCTCShepherdstown_HUC12)
+#' 
+#' # Run TADA_FindQCActivities:
+#' df <- TADA_FindQCActivities(Data_NCTCShepherdstown_HUC12)
+#' 
+#' # Find pairs for all data flagged as "QC_replicate":
+#' df_all_pairs <- TADA_PairReplicates(df)
+#' 
+#' # Find pairs for only data with ActivityTypeCode "Quality Control Sample-Field Replicate":
+#' df_fieldrep_pairs <- TADA_PairReplicates(df, type = "Quality Control Sample-Field Replicate")
+#' 
+#' # Find pairs for all data flagged as "QC_replicate" with a 10-minute time window:
```

Comment:
Suggestion to clarify in documentation: By 10-min time window, does that mean 5 mins before or after the paired sample (10 min window), or 10 mins before or 10 mins after (20 min window)?

Reply:
Good idea - right now, if time_window = 600 (10 minutes), that means it will find potential pairs 10 minutes before or 10 minutes after, so a 20 minute window. Any thoughts on a better word? maybe time_difference?

Reply:
time_difference makes sense to me. It may be helpful to say in the documentation that this is the max time difference between the paired sample and replicate. From that, the time window that the function looks at for each sample can be inferred (e.g. if the max time difference is 5 mins, then the time window that the function searches for is 10 mins (i.e., 5 mins before and 5 mins after the parent sample time).